packages/shared/client/ReplayClient.ts

```diff
@@ -611,6 +612,10 @@ export class ReplayClient implements ReplayClientInterface {
 
     const hitCounts: LineNumberToHitCountMap = new Map();
 
+    // Don't try to fetch hit counts in unloaded regions.
+    // The result might be invalid (and may get cached by a Suspense caller).
+    await this.waitForLoadedRegions(focusRange);
```

Comment:
What happens if the user drags the window wide, triggering this request, but then cancels and never hits save, or makes the region smaller again and hits save? Does this just `await` forever?

Reply:
Or will it like potentially batch a bunch of requests up and then issue them when the focus window goes wide again?

Reply:
Potentially, but it doesn't matter. Either...
1. The new focus window will contain this range, in which case it would silently complete later (and go into the Suspense cache) or
2. The new focus window won't contain this range, in which case it will continue to wait until/unless the user updates the region again, in which case it would resume from where it left off (b'c the promise is cached)

Reply:
> 1. The new focus window will contain this range, in which case it would silently complete later (and go into the Suspense cache)

Yeah, I guess I'm curious if we might queue up like 50 requests while the user is dragging the window around and then burst 50 requests when the load command finishes. That seems likely to be really inefficient, but maybe if they are all really fast then it does not matter 🤷

Reply:
Yes, we would queue up a few. I don't think this makes a big difference _in practice_ because they're generally pretty fast. It's not ideal though. TBH I'm feeling a little overwhelmed at the moment though and I'm just looking to fix things (even if not in the most efficient way)

Reply:
Yeah, I totally trust your judgement 💯

Reply:
Thank you for the discussion 🙇🏼